File: java/problem-solution/src/test/java/io/isoftware/app/TestSingleNumber.java
package io.isoftware.app;

import org.junit.Test;
import org.junit.Assert;

public class TestSingleNumber {

    @Test
    public void TestSingleNumber(){
        int[] nums = {4, 1, 4, 2, 1, 2, 5};
        Assert.assertEquals(5, SingleNumber.singleNumber(nums));

        int[] nums1 = {4};
        Assert.assertEquals(4, SingleNumber.singleNumber(nums1));

        Assert.assertEquals(10000, SingleNumber.singleNumber(new int[]{100, 100, 10000}));
    }
}

File: python/problem_soln/test_single_number.py
from single_number import singleNumber, single_number_alt


def test_single_number():
    single_number = singleNumber([1, 3, 5, 7, 9, 1, 5, 7, 9])
    assert single_number == 3

def test_single_number_alt():
    single_number = single_number_alt([1, 3, 5, 7, 9, 1, 5, 7, 9])
    assert single_number == 3
